drivers/spi/spi_ll_stm32.c

```diff
@@ -533,33 +534,34 @@ static struct spi_stm32_data spi_stm32_dev_data_##id = {		\
 	SPI_CONTEXT_INIT_SYNC(spi_stm32_dev_data_##id, ctx),		\
 };									\
 									\
-DEVICE_AND_API_INIT(spi_stm32_##id, DT_SPI_##id##_NAME, &spi_stm32_init, \
+DEVICE_AND_API_INIT(spi_stm32_##id, DT_INST_##id##_ST_STM32_SPI_LABEL,	\
+		    &spi_stm32_init,					\
 		    &spi_stm32_dev_data_##id, &spi_stm32_cfg_##id,	\
 		    POST_KERNEL, CONFIG_SPI_INIT_PRIORITY,		\
 		    &api_funcs);					\
 									\
 STM32_SPI_IRQ_HANDLER(id)
 
-#ifdef CONFIG_SPI_1
+#ifdef DT_INST_0_ST_STM32_SPI
+STM32_SPI_INIT(0)
+#endif
```

Comment:
only a suggestion...you could add here and below /* DT_INST_x_ST_STM32_SPI */ comment to be coherent with other commits

Reply:
Done